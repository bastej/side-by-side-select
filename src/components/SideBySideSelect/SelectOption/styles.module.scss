.SelectOption {
  padding: 3px 12px;
  cursor: pointer;
  &:hover {
    background-color: #eee;
  }
  &.Marked {
    background-color: #eee;
  }
  &.Disabled {
    cursor: not-allowed;
    color: #aaa;
    background-color: #ddd;
  }
}
