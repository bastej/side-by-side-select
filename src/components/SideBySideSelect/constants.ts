export const ALL_OPTIONS_SIDE = 'ALL_OPTIONS_SIDE';
export const CHOSEN_OPTIONS_SIDE = 'CHOSEN_OPTIONS_SIDE';

export const selectedItemsDefaultHeader = 'Selected';
export const allItemsDefaultHeader = 'All';
