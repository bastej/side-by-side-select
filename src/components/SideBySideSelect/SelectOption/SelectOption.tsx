import React, { MouseEventHandler, ReactNode } from 'react';
import { disabledItemDefaultTitle } from './constants';

import styles from './styles.module.scss';

export type Props = {
  onClick: MouseEventHandler<HTMLLIElement>;
  onDoubleClick: MouseEventHandler<HTMLLIElement>;
  isMarked?: boolean;
  isDisabled?: boolean;
  disabledItemTitle?: string;
  children: ReactNode;
};

const SelectOption = ({
  onClick,
  onDoubleClick,
  isMarked = false,
  isDisabled = false,
  disabledItemTitle = disabledItemDefaultTitle,
  children,
}: Props): JSX.Element => {
  const dependentProps = {
    onClick,
    onDoubleClick,
  };

  return (
    <li
      className={`${styles.SelectOption} ${isMarked ? styles.Marked : ''} ${isDisabled ? styles.Disabled : ''}`}
      {...(isDisabled ? { title: disabledItemTitle } : { ...dependentProps })}
    >
      {children}
    </li>
  );
};

export default SelectOption;
