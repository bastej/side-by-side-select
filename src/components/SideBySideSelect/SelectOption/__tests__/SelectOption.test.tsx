import React from "react";
import { render } from "@testing-library/react";
import rerender from "react-test-renderer";

import { disabledItemDefaultTitle } from "../constants";
import SelectOption from "../SelectOption";

jest.mock("../styles.module.scss", () => ({
  Disabled: "disabled",
}));

describe("<SelectOption />", () => {
  it("renders option with ", () => {
    const tree = rerender.create(
      <SelectOption onClick={jest.fn()} onDoubleClick={jest.fn()}>
        option1
      </SelectOption>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
  it("renders option without click events if options is disabled", () => {
    const tree = rerender.create(
      <SelectOption onClick={jest.fn()} onDoubleClick={jest.fn()} isDisabled>
        option1
      </SelectOption>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should set specific styles and title if options is disabled", () => {
    const root = render(
      <SelectOption onClick={jest.fn()} onDoubleClick={jest.fn()} isDisabled>
        option1
      </SelectOption>
    );

    const option = root.getByText("option1");

    expect(option.classList.contains("disabled")).toBe(true);
    expect(option.getAttribute("title")).toBe(disabledItemDefaultTitle);
  });
});
