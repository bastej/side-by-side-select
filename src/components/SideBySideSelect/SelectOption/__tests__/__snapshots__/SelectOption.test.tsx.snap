// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectOption /> renders option with  1`] = `
<li
  className="undefined  "
  onClick={[MockFunction]}
  onDoubleClick={[MockFunction]}
>
  option1
</li>
`;

exports[`<SelectOption /> renders option without click events if options is disabled 1`] = `
<li
  className="undefined  disabled"
  title="disabled"
>
  option1
</li>
`;
